hydromt/model/processes/mesh.py

```diff
@@ -166,11 +166,23 @@ def create_mesh2d_from_mesh(
     # Check if 2d mesh file else throw error
     if grid.topology_dimension != 2:
         raise ValueError("Grid in mesh file for create_mesh2d is not 2D.")
+
     # Continues with a 2D grid
     mesh2d = xu.UgridDataset(grid.to_dataset())
+    if "name" not in mesh2d.ugrid.grid.attrs:
+        mesh2d.ugrid.grid.attrs["name"] = grid.name
+    else:
+        u_name = mesh2d.ugrid.grid.attrs["name"]
+        if grid.name != u_name:
+            logger.warning(
+                f"Grid name `{grid.name}` does not match mesh2d name `{u_name}`. "
+                "Setting mesh2d name to grid name."
+            )
+            mesh2d.ugrid.grid.attrs["name"] = grid.name
```

Comment:
Is this all necessary? mesh2d is created directly from grid.to_dataset so would name differ? Especially since it seems grid.name always exist?